sbioapputils/app_runner/workflow_utils.py

```diff
@@ -0,0 +1,123 @@
+import argparse
+import os
+import warnings
+import yaml
+
+
+def _parse_workflow(request):
+    """Helper function to parse the workflow configuration."""
+    if request.get('workflow_name'):
+        workflow_loc = f"app/{request['workflow_name']}.yml"
+    else:
+        workflow_loc = "app/workflow.yml"
+        
+    with open(workflow_loc, "r") as stream:
+        try:
+            yaml_dict = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+            
+    name = yaml_dict['name']
+    stages = yaml_dict['stages']
+    parameters = yaml_dict['parameters']
+    return name, stages, parameters
+
+
+def dir_path(string):
+    """Function to check if the specified string is a directory."""
+    if os.path.isdir(string):
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+def validate_config(request, job_id):
+    """Function to validate the request config."""
+    name, stages, parameters = _parse_workflow(request)
+    for key, value in request['input_files'].items():
+        request[key] = value
+    request['job_id'] = job_id
+
+    # Check all required parameters are provided
+    no_default = []
+    no_type = []
+    wrong_data_types = []
+    invalid_value = []
+
+    for key in parameters.keys():
+
+        # Check if default is present
+        if not parameters[key].get('default'):
+            no_default.append(key)
+        elif key not in request:
+            request[key] = parameters[key]['default']
```

Comment:
I find it weird that the validation function accepts request which includes parameters and the request is validated by these parameters + parameters is validated inside the function as well.
validation function should accept 2 things: the thing to validate, validation rules, and it should validate only *the thing to validate*. if we want to validate the validation rules, we should do it somewhere else

Reply:
Changed this.